xapian-core/cluster/cluster.cc

```diff
@@ -252,15 +252,12 @@ Point::initialize(TermListGroup &tlg, const Document &doc_) {
     }
 }
 
-void
-Centroid::set_to_point(Point &p) {
-    LOGCALL_VOID(API, "Centroid::set_to_point()", p);
-    magnitude = 0;
+Centroid::Centroid(Point &p) {
     for (TermIterator it = p.termlist_begin(); it != p.termlist_end(); ++it) {
 	termlist.push_back(Wdf(*it, 1));
```

Comment:
The Wdf struct was something that we were using before for having an iterator over a document, so that we could get the wdf while computing distances. Now that we precompute the TF-IDF scores, it's only used to lookup the 'term' within the wdf.
Within the `PointType` class, we have a `vector<Wdf> termlist` and `map<string, double> values`. If we remove `termlist` from `PointType`, can I use `values` to look for terms?
It seems to reduce the redundancy within the class.